weldx/core.py

```diff
@@ -601,6 +602,52 @@ def interp_time(
         ts._interp_counter = self._interp_counter + 1
         return ts
 
+    def plot(
+        self,
+        time: Union[pd.TimedeltaIndex, pint.Quantity] = None,
+        axes: plt.Axes = None,
+        title: str = None,
+        data_name: str = "values",
+        **mpl_kwargs,
+    ) -> plt.Axes:
+        """Plot the `TimeSeries`.
+
+        Parameters
+        ----------
+        time :
+            The points in time that should be plotted. This is an optional parameter for
+            discrete `TimeSeries` but mandatory for expression based TimeSeries.
+        axes :
+            An optional matplotlib axes object
+        title :
+            The title of the plot
+        data_name :
+            Name of the data that will appear in the y-axis label
+        mpl_kwargs :
+            Key word arguments that are passed to the matplotlib plot function
+
+        Returns
+        -------
+        matplotlib.pyplot.Axes :
+            The matplotlib axes object that was used for the plot
+
+        """
+        if axes is None:
+            _, axes = plt.subplots()
+        if self.is_expression or time is not None:
+            return self.interp_time(time).plot(
+                axes=axes, title=title, data_name=data_name, **mpl_kwargs
+            )
+
+        axes.plot(self.time.values / 1e9, self._data.data.m, **mpl_kwargs)
+        axes.set_title(title)
```

Comment:
does passing `None` here remove an existing title?
in that case I would prefer having the plot function only dealing with the line and labels, leaving title etc. to other functions (so that it is easier to add TimeSeries to existing axes)

Reply:
Well, this is easy to catch with a single `if`. I will add the check.

Reply:
Removed the parameter as discussed verbally.